temporal-sdk/src/main/java/io/temporal/internal/statemachines/WorkflowStateMachines.java

```diff
@@ -416,16 +435,28 @@ private void handleSingleMessage(Message message) {
     stateMachine.handleMessage(message);
   }
 
-  private void handleSingleEvent(HistoryEvent event, boolean hasNextEvent) {
+  private void handleSingleEvent(HistoryEvent event, boolean lastTask, boolean hasNextEvent) {
     if (isCommandEvent(event)) {
       handleCommandEvent(event);
       return;
     }
 
+    // This definition of replaying here is that we are no longer replaying as soon as we
+    // see new events that have never been seen or produced by the SDK.
```

Comment:
Nice :+1: glad we're consistent now

Reply:
Yeah I just took the comment straight from core

Reply:
Hehe I noticed :)